frontend/packages/console-shared/src/hooks/useTableColumns.ts

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+import { useUserSettingsCompatibility } from './useUserSettingsCompatibility';
+import { COLUMN_MANAGEMENT_LOCAL_STORAGE_KEY } from '../constants';
+
+type TableColumnsType = {
+  [id: string]: string[];
+};
+
+export const COLUMN_MANAGEMENT_CONFIGMAP_KEY = 'console.tableColumns';
+
+export const useTableColumns = (
+  id: string,
+): [string[], (selectedColumns: Set<string>) => void, boolean] => {
+  const [tableColumns, setTableColumns, loaded] = useUserSettingsCompatibility<TableColumnsType>(
+    COLUMN_MANAGEMENT_CONFIGMAP_KEY,
+    COLUMN_MANAGEMENT_LOCAL_STORAGE_KEY,
+    {},
+    true,
+  );
+  const updateTableColumn = React.useCallback(
+    (selectedColumns: Set<string>) => {
+      setTableColumns((prevState) => {
+        return { ...prevState, [id]: [...selectedColumns] };
+      });
+    },
+    [id, setTableColumns],
+  );
+  return [loaded ? tableColumns[id] : [], updateTableColumn, loaded];
```

Comment:
nit: loaded is checked twice. Once here and once inside every call in individual components. Can we merge these 2 while we check so that we don't have to hold the loaded value in the component. Then we could also get rid of  statements like
```const [state, , loaded] to const [state];```